File: Main/src/main/java/com/bihe0832/readhub/test/base/BaseTestFragment.java
package com.bihe0832.readhub.test.base;

import android.content.Intent;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import com.bihe0832.readhub.R;
import com.ottd.libs.framework.fragment.BaseBackFragment;

import java.util.List;

public abstract class BaseTestFragment extends BaseBackFragment {
    private RecyclerView mRecy;
    private TestPagerAdapter mAdapter;

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        View view = inflater.inflate(R.layout.com_bihe0832_test_tab_fragment, container, false);
        initView(view);
        return view;
    }

    private void initView(View view) {

        mAdapter = new TestPagerAdapter(_mActivity);
        mAdapter.setDatas(getDataList());
        mRecy = (RecyclerView) view.findViewById(R.id.test_recy);
        mRecy.setHasFixedSize(true);
        LinearLayoutManager manager = new LinearLayoutManager(_mActivity);
        mRecy.setLayoutManager(manager);
        mRecy.setAdapter(mAdapter);
    }

    @Override
    public void onDestroyView() {
        super.onDestroyView();
    }

    public abstract List<TestItem> getDataList();

    protected void sendInfo(String title, String content){
        Intent sendIntent = new Intent();
        sendIntent.setAction(Intent.ACTION_SEND);
        sendIntent.putExtra(Intent.EXTRA_TEXT, content);
        sendIntent.setType("text/plain");
        startActivity(Intent.createChooser(sendIntent, title));
    }

}

File: Main/src/main/java/com/bihe0832/readhub/test/base/BaseTestFragment.kt
package com.bihe0832.readhub.test.base

import android.content.Intent
import android.os.Bundle
import android.support.v7.widget.LinearLayoutManager
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import com.bihe0832.readhub.R
import com.ottd.libs.framework.fragment.BaseBackFragment
import kotlinx.android.synthetic.main.com_bihe0832_test_tab_fragment.*

abstract class BaseTestFragment : BaseBackFragment() {
    private val pagerAdapter: TestPagerAdapter by lazy { TestPagerAdapter(_mActivity) }

    abstract fun getDataList(): List<TestItem>

    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        val view = inflater.inflate(R.layout.com_bihe0832_test_tab_fragment, container, false)
        initView(view)
        return view
    }

    private fun initView(view: View) {
        pagerAdapter.setData(getDataList())
        test_recy.apply {
            setHasFixedSize(true)
            layoutManager = LinearLayoutManager(_mActivity)
            adapter = pagerAdapter
        }
    }

    protected fun sendInfo(title: String, content: String) {
        Intent().apply {
            action = Intent.ACTION_SEND
            putExtra(Intent.EXTRA_TEXT, content)
            type = "text/plain"
        }.let { startActivity(Intent.createChooser(it, title)) }
    }

}
